### ISSUE TITLE
vsphere-iso format ovf no longer works?

### ISSUE THREAD

**Issue by lamw** [hashicorp/packer#8752]
#### Overview of the Issue

In the vsphere-iso documentation, it mentions Export option should still work and when specifying `format=ovf`, unknown configuration key error is thrown. How do we export the image to OVF using the new vsphere-iso builder?

#### Reproduction Steps

./packer_darwin_amd64 build -var-file=photon-builder.json -var-file=photon-version.json vsphere-iso-packer-build.json

### Packer version

Packer v1.5.3-dev

### Simplified Packer Buildfile

https://gist.github.com/lamw/b4ea5dc271da02ec2289205db7198641

### Operating system and Environment details

MacOS Catalina

### Log Fragments and crash.log files

```
2020/02/16 05:58:12 Build 'vsphere-iso' prepare failure: 1 error occurred:
1 error occurred:
	* unknown configuration key: "format"; raws is []interface {}{map[string]interface {}{"CPUs":"2", "RAM":"4096", "boot_command":[]interface {}{"<esc><wait>", "vmlinuz initrd=initrd.img root=/dev/ram0 loglevel=3 ks=http://{{ .HTTPIP }}:{{ .HTTPPort }}/photon-kickstart.json photon.media=cdrom", "<enter>"}, "boot_wait":"10s", "cluster":"Supermicro-Cluster", "datacenter":"Primp-Datacenter", "datastore":"sm-vsanDatastore", "disk_size":"12288", "disk_thin_provisioned":true, "firmware":"bios", "format":"ovf", "guest_os_type":"otherGuest", "http_directory":"http", "insecure_connection":true, "iso_checksum":"93d0cde8da51f9208713d895b5b85b86980d2a72e710f55f0e65bc82c299dd9a7ebedc8f30d5f4d18c1a389c76a961e8a14b02416692204d31d77e1e4792f37d", "iso_checksum_type":"sha512", "iso_paths":[]interface {}{"[sm-vsanDatastore] packer_cache/a3b59ec87af86d5595419065d04f73f156ad4cfe.iso"}, "iso_url":"http://dl.bintray.com/vmware/photon/3.0/GA/iso/photon-3.0-26156e2.iso", "network":"VM Network", "network_card":"vmxnet3", "password":"VMware1!", "shutdown_command":"/sbin/shutdown -h now", "shutdown_timeout":"1000s", "ssh_password":"VMware1!", "ssh_port":22, "ssh_username":"root", "username":"administrator@vsphere.local", "vcenter_server":"192.168.30.200", "vm_name":"PhotonOS_Appliance", "vm_version":"13"}, map[string]interface {}{"packer_build_name":"vsphere-iso", "packer_builder_type":"vsphere-iso", "packer_debug":false, "packer_force":false, "packer_on_error":"", "packer_template_path":"/Volumes/Storage/Software/photonos-appliance/photon.json", "packer_user_variables":map[interface {}]interface {}{"builder_host":"192.168.30.200", "builder_host_cluster":"Supermicro-Cluster", "builder_host_datacenter":"Primp-Datacenter", "builder_host_datastore":"sm-vsanDatastore", "builder_host_folder":"Dev", "builder_host_password":"VMware1!", "builder_host_portgroup":"VM Network", "builder_host_username":"administrator@vsphere.local", "description":"PhotonOS Appliance", "guest_password":"VMware1!", "guest_username":"root", "iso_checksum":"93d0cde8da51f9208713d895b5b85b86980d2a72e710f55f0e65bc82c299dd9a7ebedc8f30d5f4d18c1a389c76a961e8a14b02416692204d31d77e1e4792f37d", "iso_checksum_type":"sha512", "iso_url":"http://dl.bintray.com/vmware/photon/3.0/GA/iso/photon-3.0-26156e2.iso", "numvcpus":"2", "ovftool_deploy_vcenter":"192.168.30.200", "ovftool_deploy_vcenter_password":"VMware1!", "ovftool_deploy_vcenter_username":"administrator@vsphere.local", "photon_ovf_template":"photon.xml.template", "ramsize":"4096", "version":"0.1.0", "vm_name":"PhotonOS_Appliance"}}}
```

**Comment by jhawk28** [comment]
I don't see a format option in the documentation: https://packer.io/docs/builders/vsphere-iso.html
The only export format option is in the vmware-iso builder: https://packer.io/docs/builders/vmware-iso.html#format

**Comment by sylviamoss** [comment]
Hey @lamw, thanks for reaching out! I must agree with @jhawk28, `vsphere` builders don't support `format` config. 
Can you tell me your context and why you need the format config? I don't have much knowledge on vsphere builder itself right now but I want to understand if this config is something we could add.

**Comment by lamw** [comment]
@sylviamoss - Hm, when looking at the [vsphere-iso docs](https://packer.io/docs/builders/vsphere-iso.html), at the top it mentions the following:

```
Configuration Reference
There are many configuration options available for this builder. In addition to the items listed here, you will want to look at the general configuration references for ISO, HTTP, Floppy, Boot, Driver, Hardware, Output, Run, Shutdown, Communicator, Tools, vmx, Export, configuration references, which are necessary for this build to succeed and can be found further down the page.
```

I had rad this to assume `Export` was supported similiar to that of vmware-iso builder. The use case is that I'd like to be able to export the image I just built into an OVF which I then process further. Here's a complete reference example https://github.com/lamw/photonos-appliance which I'd like to convert over to vsphere-iso as it doesn't require any tweaks to the ESXi host which not all customers have the luxary of doing.

FYI - I work for VMware

**Comment by jhawk28** [comment]
It would probably be useful to add an import/export enhancement to the two vsphere plugins. vsphere-clone could use an import ova/ovf mechanism. vsphere-iso and vsphere-clone could use an ovf/ova export.

govc gives an example on how to export here: https://github.com/vmware/govmomi/blob/master/govc/export/ovf.go
and import here: https://github.com/vmware/govmomi/tree/master/govc/importx

**Comment by lamw** [comment]
Definitely would be useful to have export at a minimal, I guess I wrongfully assumed functionality wise between vmware-iso and vsphere-iso builder. This regression wouldn't allow me to consume the current builder as-is as most will build a template and then export that out.

**Comment by sylviamoss** [comment]
@lamw I was wondering if you can get the ovf format by using [vsphere post-processor](https://packer.io/docs/post-processors/vsphere.html).
Anyways, I agree we can implement the export mechanism for this builder. I see that there's an open PR already 👍 let's see and maybe you can test the binaries to make sure it works as expected.

**Comment by lamw** [comment]
I'll wait for the PR to get added as I was only interested in updating the builder from vmware to vsphere without changing the rest of the code I've got.

**Comment by nywilken** [comment]
Hello :wave: support for exporting a VM to an ovf template file has been merged in  #8764. The feature will be released in v1.5.5 of  Packer. But if you would like to gain access to the feature sooner it will be available in the next [nightly release](https://github.com/hashicorp/packer/releases/tag/nightly)
